src/osd/OSD.cc

```diff
@@ -9322,6 +9325,18 @@ void OSD::ShardedOpWQ::wake_pg_waiters(spg_t pgid)
   }
 }
 
+void OSD::ShardedOpWQ::prime_splits(const set<spg_t>& pgs)
+{
+  dout(20) << __func__ << " " << pgs << dendl;
+  for (auto pgid : pgs) {
+    unsigned shard_index = pgid.hash_to_shard(shard_list.size());
+    ShardData* sdata = shard_list[shard_index];
+    Mutex::Locker l(sdata->sdata_op_ordering_lock);
+    ShardData::pg_slot& slot = sdata->pg_slots[pgid];
+    slot.waiting_for_pg = true;
```

Comment:
It's not wrong, but the ShardedOpWQ sure is grossly aware of what's going on with PGs at this point. :(

Reply:
yes.  the shardedwq becomes the complicated box that everything funnels through.  my hope is that this will continue to be consolidated into an overall OSDShard-type concept (not just the queue), and ultimately be simpler to understand and replace in a seastar world than the hodge-podge of threads and locks we had before.